circuit/src/ecc/generic/mod.rs

```diff
@@ -1,2 +1,1754 @@
 /// Native implementation
 pub mod native;
+
+use self::native::EcPoint;
+use super::{
+	EccAddConfig, EccBatchedMulConfig, EccDoubleConfig, EccMulConfig, EccTableSelectConfig,
+	EccUnreducedLadderConfig,
+};
+use crate::{
+	gadgets::{
+		bits2integer::{Bits2IntegerChipset, Bits2IntegerChipsetConfig},
+		main::SelectChipset,
+	},
+	integer::{
+		AssignedInteger, IntegerAddChip, IntegerDivChip, IntegerMulChip, IntegerReduceChip,
+		IntegerSubChip, UnassignedInteger,
+	},
+	params::{ecc::EccParams, rns::RnsParams},
+	utils::{assigned_as_bool, be_assigned_bits_to_usize},
+	Chip, Chipset, CommonConfig, FieldExt, UnassignedValue,
+};
+use halo2::{
+	circuit::{AssignedCell, Layouter},
+	halo2curves::ff::PrimeField,
+	halo2curves::CurveAffine,
+	plonk::Error,
+};
+use itertools::Itertools;
+use std::marker::PhantomData;
+
+/// Structure for the UnassignedEcPoint
+#[derive(Clone, Debug)]
+pub struct UnassignedEcPoint<
+	C: CurveAffine,
+	N: FieldExt,
+	const NUM_LIMBS: usize,
+	const NUM_BITS: usize,
+	P,
+	EC,
+> where
+	P: RnsParams<C::Base, N, NUM_LIMBS, NUM_BITS> + RnsParams<C::ScalarExt, N, NUM_LIMBS, NUM_BITS>,
+	EC: EccParams<C>,
```

Comment:
Looks like `EC: EccParams<C>,` is not needed for UnassignedEcPoint?

Reply:
Yes, fixed.